### PR TITLE
[KYUUBI #2494] Support executeBatch with KyuubiBatchStatement

### LINKED ISSUE #2494: [FEATURE] Support executeBatch in Kyuubi JDBC
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

We always set some configs or do some pre-check ddl before running one or more dml, so it can be more effective if we can treat them as a batch and execute one by one at engien side.

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Close #2494 
Support to execute multiple queries.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-hive-jdbc/src/test/java/org/apache/kyuubi/jdbc/hive/KyuubiBatchStatementTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+package org.apache.kyuubi.jdbc.hive;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class KyuubiBatchStatementTest {
+    @Test
+    public void testSplitSemiColon() {
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/test/java/org/apache/kyuubi/jdbc/hive/KyuubiBatchStatementTest.java:28]: let's add some complex cases, e.g. `SELECT regexp(col, '[abc;]+')`

**Comment by turboFei** [comment]
Cc @ulysses-you

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBatchStatement.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.*;
+import org.apache.hive.service.rpc.thrift.TCLIService;
+import org.apache.hive.service.rpc.thrift.TSessionHandle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KyuubiBatchStatement extends KyuubiStatement {
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBatchStatement.java:29]: I'm wondering why not merge `KyuubiBatchStatement` into `KyuubiStatement`?

Comment (turboFei) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBatchStatement.java:29]: I tried that, it is very complex.

Now the resultSet is couple with KyuubiStatement.

one statementHandle per KyuubiStatement.

Comment (turboFei) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBatchStatement.java:29]: If we did want merge it into KyuubiStatement, we need modify much code and make it difficult to maintain.

**Comment by ulysses-you** [comment]
I thought about it. Ideally, `executeBatch` have two implementations:
1. split statements at client side which is this pr did
2. split statements at server side

I think the advantage of 2 is we can support with hive-jdbc and rest-api easily. what do you think about ? @turboFei @yaooqinn

**Comment by turboFei** [comment]
> split statements at server side

+1

**Comment by cfmcgrady** [comment]
`at server side` means make the server stateful?

**Comment by turboFei** [comment]
> at server side means make the server stateful?

Not sure

**Comment by turboFei** [comment]
close this one, see #3392 3392

**Pull request closed without merging**